### PR TITLE
add spinner in install plugin on homepage

### PR DESCRIPTION
**Pull request by lilyanB**
goal: add spinner during installation of plugin.

Local installation is very quick. You can add this code for testing purposes:
```
      setTimeout(() => {
        console.log('timeout');
        setIsLoading(false);
      }, 10000);
```

### CODE REVIEW

**Review by pivilartisant — APPROVED** [review]

File: web/massastation/src/pages/Index/Index.tsx [hunk 1]
```diff
@@ -123,22 +127,37 @@ export function Index() {
...
+                          )
+                        }
+                        onClickInactive={handleInstallPlugin}
+                      />,
+                    ]
+                  : [
+                      <div>
+                        <WalletInactive />
+                        <Button
+                          type="submit"
```
Comment (mazmassa) [inline web/massastation/src/pages/Index/Index.tsx:152]: The button is set to be always disabled, isn't it... I don't think you need a type submit!?

File: web/massastation/src/pages/Index/Index.tsx [hunk 2]
```diff
@@ -123,22 +127,37 @@ export function Index() {
...
+                          customClass="mt-3"
```
Comment (mazmassa) [inline web/massastation/src/pages/Index/Index.tsx:153]: **customClass** and with a margin-top. I think we should rethink a bit that!

File: web/massastation/src/pages/Index/Index.tsx [hunk 3]
```diff
@@ -123,22 +127,37 @@ export function Index() {
...
+                          window.open(
+                            '/plugin/massa-labs/massa-wallet/web-app/index',
+                            '_blank',
+                          )
+                        }
+                        onClickInactive={handleInstallPlugin}
+                      />,
+                    ]
+                  : [
+                      <div>
```
Comment (mazmassa) [inline web/massastation/src/pages/Index/Index.tsx:149]:
```suggestion
                      <>
```

File: web/massastation/src/pages/Index/Index.tsx [hunk 4]
```diff
@@ -69,11 +71,13 @@ export function Index() {
...
   useEffect(() => {
     setRefreshPlugins(refreshPlugins + 1);
-  }, [pluginWalletIsInstalled]);
+  }, [pluginWalletIsInstalled, isLoading]);
 
   function handleInstallPlugin() {
     try {
+      setIsLoading(true);
       mutate({});
+      setIsLoading(false);
```
Comment (mazmassa) [inline web/massastation/src/pages/Index/Index.tsx:80]: Mutate is async... This should be false on the isSuccess/isError step.

**Comment by mazmassa** [comment]
> goal: add spinner during installation of plugin.
> 
> Local installation is very quick. You can add this code for testing purposes:
> 
> ```
>       setTimeout(() => {
>         console.log('timeout');
>         setIsLoading(false);
>       }, 10000);
> ```

You don't need that.
Just add in the Server.js file, line 181, that:

}, { timing: 4000});
 and your request will take 4s!

File: web/massastation/src/pages/Index/Index.tsx (hunk 4 above)
Comment (lilyanB) [reply web/massastation/src/pages/Index/Index.tsx]: done

File: web/massastation/src/pages/Index/Index.tsx (hunk 3 above)
Comment (lilyanB) [reply web/massastation/src/pages/Index/Index.tsx]: done

File: web/massastation/src/pages/Index/Index.tsx (hunk 2 above)
Comment (lilyanB) [reply web/massastation/src/pages/Index/Index.tsx]: oh! sorry... It's du to copy/past...
Done

File: web/massastation/src/pages/Index/Index.tsx (hunk 1 above)
Comment (lilyanB) [reply web/massastation/src/pages/Index/Index.tsx]: done

**Comment by lilyanB** [comment]
> > goal: add spinner during installation of plugin.
> > Local installation is very quick. You can add this code for testing purposes:
> > ```
> >       setTimeout(() => {
> >         console.log('timeout');
> >         setIsLoading(false);
> >       }, 10000);
> > ```
> 
> You don't need that. Just add in the Server.js file, line 181, that:
> 
> }, { timing: 4000}); and your request will take 4s!

Thanks for your tips

**Review by mazmassa — APPROVED** [review]

**Pull request merged**